x-pack/plugins/ingest_manager/common/services/is_agent_upgradeable.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import semver from 'semver';
+import { Agent } from '../types';
+
+export function isAgentUpgradeable(agent: Agent, kibanaVersion: string) {
```

Comment:
_I realize this is draft, just noting while reading:_ even though this is currently a simple "agent < kibana" check, it'd be nice to have a test with a few examples to confirm our thinking.

Reply:
this function will be removed, its temporary until agent reports upgradeability.